src/index.js

```diff
@@ -116,7 +123,14 @@ class CookieConsent extends Component {
     // remove listener if set
     window.removeEventListener("scroll", this.handleScroll);
 
-    Cookies.set(cookieName, cookieValue, { expires: expires, ...extraCookieOptions });
+    if (sameSite === SAMESITE_OPTIONS.NONE) {
+      Cookies.set(cookieName, cookieValue, { expires, sameSite, secure: true, ...extraCookieOptions });
+
+      // Fallback for older browsers where can not set SameSite=None, SEE: https://web.dev/sameSite-cookie-recipes/#handling-incompatible-clients
+      Cookies.set(cookieName, cookieValue, { expires, secure: true, ...extraCookieOptions });
```

Comment:
This won't work unfortunately. You have to give both cookies a different name (as in the example) and check for the new one first and the legacy one second whilst retrieving.

Reply:
Ok I could use these helper functions from Chromium: https://www.chromium.org/updates/same-site/incompatible-clients to detect incompatible browser that do not support samesite: none, do you mind if I copy over the methods into a helper file?

Reply:
Oh my man, I didn't see your comment!

The solution provided (by setting 2 cookies) is perfectly elegant imo.
Adding extra code would bloat the package i.m.o

The thing is, the way you implemented it was not correct. You did:

- set cookie with all options
- set same cookie without options
- retrieve cookie (this will get either one)

What you should have done is:

- set cookie with all options
- set a cookie with a different name (the author uses the -legacy suffix) without the options
- read the normal cookie and if that isn't set (because the browser was incompatible) read the legacy cookie.


Take a look at how I fixed it here: https://github.com/Mastermindzh/react-cookie-consent/pull/68/commits/a64cbc28f9458a4bf4cc89590eded31f78afce2d#diff-1fdf421c05c1140f6d71444ea2b27638R167-R197